sormas-backend/src/main/resources/sql/sormas_schema.sql

```diff
@@ -7940,4 +7940,329 @@ ALTER TABLE task_history ADD COLUMN travelentry_id bigint;
+        vaccinationinfo.vaccinemanufacturer, vaccinationinfo.othervaccinemanufacturer, vaccinationinfo.vaccinationinfosource, vaccinationinfo.vaccineinn,
+        vaccinationinfo.vaccinebatchnumber, vaccinationinfo.vaccineuniicode, vaccinationinfo.vaccineatccode, null AS pregnant, null AS trimester, null AS healthconditions_id,
+        coalesce(events.startdate, events.enddate, events.reportdatetime) AS relevancedate
+    FROM person
+             LEFT JOIN eventparticipant ON eventparticipant.person_id = person.id
+             LEFT JOIN events ON eventparticipant.event_id = events.id
+             LEFT JOIN location ON events.eventlocation_id = location.id
+             LEFT JOIN vaccinationinfo ON eventparticipant.vaccinationinfo_id = vaccinationinfo.id
+    WHERE vaccinationinfo.vaccination = 'VACCINATED' AND eventparticipant.deleted = false
+    ORDER BY person.id, events.disease, relevancedate DESC
+);
+
+DROP TABLE IF EXISTS tmp_vaccinated_persons;
+CREATE TEMP TABLE tmp_vaccinated_persons AS
+SELECT DISTINCT ON (person_id, disease) person_id,
+                                        disease, diseasedetails, reportdate, reportinguser_id, responsibleregion_id, responsibledistrict_id, responsiblecommunity_id,
+                                        firstvaccinationdate, lastvaccinationdate, vaccinationdoses, vaccinename, othervaccinename, vaccinemanufacturer, othervaccinemanufacturer,
+                                        vaccinationinfosource, vaccineinn, vaccinebatchnumber, vaccineuniicode, vaccineatccode, pregnant, trimester, healthconditions_id,
+                                        case_id, nextval('entity_seq') AS immunization_id, relevancedate
+FROM tmp_vaccinated_entities
+ORDER BY person_id, disease, relevancedate DESC;
+
+/* Step 2: Create a new immunization entity for each person-disease combination */
+INSERT INTO immunization
+(
+    id, uuid, disease, diseasedetails, person_id, reportdate, reportinguser_id, immunizationstatus, meansofimmunization, immunizationmanagementstatus, responsibleregion_id,
+    responsibledistrict_id, responsiblecommunity_id, startdate, enddate, numberofdoses, relatedcase_id, changedate, creationdate
+)
+SELECT
+    immunization_id, generate_base32_uuid(), disease, diseasedetails, person_id, reportdate, reportinguser_id, 'ACQUIRED', 'VACCINATION', 'COMPLETED',
+    responsibleregion_id, responsibledistrict_id, responsiblecommunity_id, firstvaccinationdate, lastvaccinationdate, vaccinationdoses,
+    case_id, now(), now()
+FROM tmp_vaccinated_persons;
+
+/* Step 3: Create a new vaccination entity for each immunization start and date (or for each immunization without a start or end date) */
+CREATE OR REPLACE FUNCTION clone_healthconditions(healthconditions_id bigint)
+    RETURNS bigint
+    LANGUAGE plpgsql
+    SECURITY DEFINER AS
+$BODY$
+DECLARE new_id bigint;
+BEGIN
+    DROP TABLE IF EXISTS tmp_healthconditions;
```

Comment:
Do you think that truncating it every time is significantly faster?

Reply:
No, probably wont make a bit difference  especially since it's a temporary table. There is no point in further investigating this unless we run into an actual problem.